src/poly/commitment/verifier.rs

```diff
@@ -62,144 +64,104 @@ impl<'a, C: CurveAffine> Guard<'a, C> {
     }
 }
 
-impl<C: CurveAffine> Proof<C> {
-    /// Checks to see if an [`Proof`] is valid given the current `transcript`,
-    /// and a point `x` that the polynomial commitment `p` opens purportedly to
-    /// the value `v`.
-    pub fn verify<'a, HBase, HScalar>(
-        &self,
-        params: &'a Params<C>,
-        mut msm: MSM<'a, C>,
-        transcript: &mut Transcript<C, HBase, HScalar>,
-        x: C::Scalar,
-        mut commitment_msm: MSM<'a, C>,
-        v: C::Scalar,
-    ) -> Result<Guard<'a, C>, Error>
-    where
-        HBase: Hasher<C::Base>,
-        HScalar: Hasher<C::Scalar>,
-    {
-        // Check for well-formedness
-        if self.rounds.len() != params.k as usize {
+/// Checks to see if an [`Proof`] is valid given the current `transcript`, and a
+/// point `x` that the polynomial commitment `P` opens purportedly to the value
+/// `v`. The provided `commitment_msm` should evaluate to the commitment `P`
+/// being opened.
+pub fn verify_proof<'a, C: CurveAffine, R: Read, T: TranscriptRead<R, C>>(
+    params: &'a Params<C>,
+    mut msm: MSM<'a, C>,
+    transcript: &mut T,
+    x: C::Scalar,
+    mut commitment_msm: MSM<'a, C>,
```

Comment:
This can be merged with `msm` now that we don't scale the commitment we're opening by anything.